otter/test/cloud_client/test_cloudfeeds.py

```diff
@@ -43,3 +47,147 @@ def test_publish_autoscale_event(self):
             expected.intent,
             service_request_eqf(stub_pure_response('<this is xml>', 202)))])
         self.assertRaises(APIError, sync_perform, dispatcher, eff)
+
+
+def both_links(method):
+    """
+    Call given test method with "next" and "previous" rel
+    """
+    @wraps(method)
+    def f(self):
+        method(self, "next")
+        method(self, "previous")
+    return f
+
+
+class ReadEntriesTests(SynchronousTestCase):
+
+    entry = '<entry><summary>{}</summary></entry>'
+    feed_fmt = (
+        '<feed xmlns="http://www.w3.org/2005/Atom">'
+        '<link rel="{rel}" href="{rel_link}"/>'
+        '{entries}</feed>')
+    url = "http://url"
+    directions = {"previous": cf.Direction.PREVIOUS,
+                  "next": cf.Direction.NEXT}
+    service_type = ServiceType.CLOUD_FEEDS
+
+    def svc_intent(self, params={}):
+        return service_request(
+            self.service_type, "GET",
+            self.url, params=params, json_response=False).intent
+
+    def feed(self, rel, link, summaries):
+        return self.feed_fmt.format(
```

Comment:
This strategy is going to quote all kinds of stuff improperly.  It might be good enough for this use-case, but, pro tip:

```
>>> from twisted.web.template import Tag, flattenString
>>> flattenString(None, Tag('feed')(xmlns="http://www.w3.org/2005/Atom")(Tag('link')(rel="something",
href="something"))).result
'<feed xmlns="http://www.w3.org/2005/Atom"><link href="something" rel="something" /></feed>'
```